### DISCUSSION TITLE
Running YOLOv5 or YOLOv8 on Boston Dynamic's Spot Dog Robot

### DISCUSSION THREAD

**Discussion by CyrusAddy** [ultralytics/ultralytics#4244] [Docs]
Hello,

Has anyone ran a real-time object detection on Spot Dog using YOLOV5 or YOLOv8? I need some guidance on how to do that. Thank you.

**Comment by glenn-jocher** [comment]
@CyrusAddy hello!

Running real-time object detection on Boston Dynamics' Spot using YOLOv8 is an exciting application! While I'll focus on YOLOv8, the process is quite similar for both versions. Here's a high-level overview:

1. **Setup**: Ensure Spot's onboard computer meets the hardware requirements for running YOLOv8, including a compatible GPU for real-time inference if needed.

2. **Installation**: Install YOLOv8 on Spot's computer by following the installation instructions in our documentation.

3. **Model Selection**: Choose a pre-trained YOLOv8 model or train your own on a relevant dataset depending on your detection needs.

4. **Integration**: Develop a software interface between Spot's sensory system (cameras, etc.) and YOLOv8 to feed live video frames into the model for detection.

5. **Inference**: Use the Predict mode to run real-time object detection on the video feed.

6. **Action**: Program Spot to respond to the detection outputs according to your application's requirements.

For detailed guidance on each step, please refer to our documentation. If you encounter specific issues or have questions during implementation, feel free to raise them in the repo, and we'll be happy to assist. Good luck with your project! 🤖🐕
